k8s/controllers/federation/catalog_controller.go

```diff
@@ -47,6 +50,9 @@ func (r *CatalogReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	diagnostic.InfoWithCtx(ctrlLog, ctx, "Reconciling Catalog", "Name", req.Name, "Namespace", req.Namespace)
 	catalog := &federationv1.Catalog{}
+	resourceK8SId := catalog.GetNamespace() + "/" + catalog.GetName()
+	operationName := fmt.Sprintf("%s/%s", constants.CatalogOperationNamePrefix, constants.ActivityOperation_Write)
```

Comment:
You need to use DeletionTimestamp.IsZero() to determine the delete and write action.